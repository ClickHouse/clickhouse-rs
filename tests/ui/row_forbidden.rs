#[derive(clickhouse::Row)]
enum Enum {
    A(i32),
    B(String),
}

#[derive(clickhouse::Row)]
union Union {
    a: i32,
    b: u64,
}

#[derive(clickhouse::Row)]
pub struct UnitStruct;

#[derive(clickhouse::Row)]
pub struct EmptyStruct {}

fn main() {}
