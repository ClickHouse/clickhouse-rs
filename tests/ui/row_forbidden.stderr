error: `Row` can only be derived for structs
 --> tests/ui/row_forbidden.rs:2:6
  |
2 | enum Enum {
  |      ^^^^

error: `Row` can only be derived for structs
 --> tests/ui/row_forbidden.rs:8:7
  |
8 | union Union {
  |       ^^^^^

error: `Row` cannot be derived for unit or empty structs
  --> tests/ui/row_forbidden.rs:14:12
   |
14 | pub struct UnitStruct;
   |            ^^^^^^^^^^

error: `Row` cannot be derived for unit or empty structs
  --> tests/ui/row_forbidden.rs:17:12
   |
17 | pub struct EmptyStruct {}
   |            ^^^^^^^^^^^
